server/src/main/java/org/opensearch/search/query/ConcurrentQueryPhaseSearcher.java

```diff
@@ -101,4 +101,26 @@ private static boolean searchWithCollectorManager(
     public AggregationProcessor aggregationProcessor(SearchContext searchContext) {
         return aggregationProcessor;
     }
+
+    private static <T extends Exception> void rethrowCauseIfPossible(RuntimeException re, SearchContext searchContext) throws T {
+        // Rethrow exception if cause is null
+        if (re.getCause() == null) {
+            throw re;
+        }
+
+        // Unwrap the RuntimeException and ExecutionException from Lucene concurrent search method and rethrow
+        if (re.getCause() instanceof ExecutionException || re.getCause() instanceof InterruptedException) {
+            Throwable t = re.getCause();
+            if (t.getCause() != null) {
+                throw (T) t.getCause();
+            }
+        }
+
+        // Rethrow any unexpected exception types
+        throw new QueryPhaseExecutionException(
+            searchContext.shardTarget(),
+            "Failed to execute concurrent segment search thread",
+            re.getCause()
+        );
+    }
```

Comment:
I'm wrapping any unexpected exception types in `QueryPhaseExecutionException`, but still keeping the casting for the `ExecutionException` and `InterruptedException` cases as I am trying to preserve the `TaskCancelledException` that is returned to the user in the case of non-concurrent search here.

Reply:
It is much cleaner now, thank you @jed326

Reply:
lgtm. thanks for the changes